test/e2e/multihoming.go

```diff
@@ -636,6 +640,161 @@ var _ = Describe("Multi Homing", func() {
+					cmd.Stderr = os.Stderr
+					err := cmd.Run()
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				AfterEach(func() {
+					By("tearing down the localnet underlay")
+					Expect(teardownUnderlay(nodes)).To(Succeed())
+				})
+
+				It("can communicate over a localnet secondary network from pod to the underlay service", func() {
+					clientPodConfig := podConfiguration{
+						name:        clientPodName,
+						namespace:   f.Namespace.Name,
+						attachments: []nadapi.NetworkSelectionElement{{Name: secondaryNetworkName}},
+					}
+					kickstartPod(cs, clientPodConfig)
+
+					By("asserting the *client* pod can contact the underlay service")
+					Expect(connectToServer(clientPodConfig, underlayServiceIP, servicePort)).To(Succeed())
+				})
+
+				Context("with multi network policy blocking the traffic", func() {
+					var clientPodConfig podConfiguration
+					labels := map[string]string{"name": "access-control"}
+
+					const policyName = "allow-egress-ipblock"
+
+					BeforeEach(func() {
+						clientPodConfig = podConfiguration{
+							name:        clientPodName,
+							namespace:   f.Namespace.Name,
+							attachments: []nadapi.NetworkSelectionElement{{Name: secondaryNetworkName}},
+							labels:      labels,
+						}
+						kickstartPod(cs, clientPodConfig)
+
+						By("asserting the *client* pod can contact the underlay service before creating the policy")
+						Expect(connectToServer(clientPodConfig, underlayServiceIP, servicePort)).To(Succeed())
+					})
+
+					It("can not communicate over a localnet secondary network from pod to the underlay service", func() {
+						By("creating the multi-network policy")
+						_, err := mnpClient.MultiNetworkPolicies(clientPodConfig.namespace).Create(
+							context.Background(),
+							multiNetEgressLimitingIPBlockPolicy(
+								policyName,
+								secondaryNetworkName,
+								metav1.LabelSelector{
+									MatchLabels: labels,
+								},
+								mnpapi.IPBlock{
+									CIDR:   secondaryLocalnetNetworkCIDR,
+									Except: []string{underlayServiceIP},
+								},
+							),
+							metav1.CreateOptions{},
+						)
+						Expect(err).NotTo(HaveOccurred())
+
+						By("asserting the *client* pod cannot contact the underlay service after creating the policy")
+						Expect(connectToServer(clientPodConfig, underlayServiceIP, servicePort)).To(MatchError(ContainSubstring("Connection timeout")))
+					})
+
+					AfterEach(func() {
+						By("deleting the multi-network policy")
+						err := mnpClient.MultiNetworkPolicies(clientPodConfig.namespace).Delete(
+							context.Background(),
+							policyName,
+							metav1.DeleteOptions{},
+						)
+						Expect(err).NotTo(HaveOccurred())
+
+						By("asserting the *client* pod can contact the underlay service after deleting the policy")
+						Expect(connectToServer(clientPodConfig, underlayServiceIP, servicePort)).To(Succeed())
+					})
```

Comment:
Please move this before the test.

The common pattern for a test is:
1. define the test setup (`BeforeEach`)
2. define the test teardown (`AfterEach`)
3. define the test

Reply:
done